src/bluez/peripheral.rs

```diff
@@ -138,6 +138,17 @@ impl api::Peripheral for Peripheral {
         self.mac_address
     }
 
+    fn mtu(&self) -> u16 {
+        let services = self.services.lock().unwrap();
+        for (_, service) in services.iter() {
+            for (_, characteristic) in service.characteristics.iter() {
+                return characteristic.info.mtu.unwrap();
```

Comment:
Can this be different for different characteristics?

Reply:
In Bluetooth 5.2 and forward, yes. The reason they implemented it as a per-characteristic MTU is because of the enhanced ATT (EATT) bearer feature.

They mention it here: https://github.com/bluez/bluez/issues/199
I mention it here: https://github.com/deviceplug/btleplug/issues/246#issuecomment-1545186841

The majority of devices on the market are going to be using the unenhanced version (single MTU per connection), so I don't understand why bluez didn't add support for that first.

In any case, the implementation was based on how SimpleBle did it: https://github.com/OpenBluetoothToolbox/SimpleBLE/blob/614a5af35cdd22ba08318cb776795ecd2da8c389/simpleble/src/backends/linux/PeripheralBase.cpp#L57

Reply:
Okay, seems reasonable given the API limitations.